### PR TITLE
draft: libc support for hexagon linux

### PR DESCRIPTION
**Pull request by androm3da**
**this PR not in a useful state for review**

### CODE REVIEW

File: libc/include/llvm-libc-types/fenv_t.h [hunk 1]
```diff
@@ -25,6 +25,8 @@ typedef struct {
 } fenv_t;
 #elif defined(__riscv)
```
Comment (lntue) [inline libc/include/llvm-libc-types/fenv_t.h:26]: merge with `__riscv` instead.

**Comment by androm3da** [comment]
Closing: this PR is too far out of date to be useful.  Hopefully it will get revisited in the not-too-distant future.

**Comment by nickdesaulniers** [comment]
FWIW, I spoke with @apazos at the '24 US LLVM Dev conf about this, which led to the email about qemu-system support.

We're going to be revisiting the buildbots in H1'25; I'm looking to move most of our CI to cross compilation+testing in qemu.  We might not need qemu-system support just yet.

At that point, wiring up more buildbots for additional architectures should become more straightforward for us.

**Comment by androm3da** [comment]
> FWIW, I spoke with @apazos at the '24 US LLVM Dev conf about this, which led to the email about qemu-system support.

Ok, gotcha.  Buildbots are not a major factor in this particular PR succeeding or failing IMO.  This PR was very much incomplete - missing some significant work in crt0.  Compiler was generating a preamble that couldn't coexist with the inline asm in `_start`.  This problem might not be tough to solve, but it was probably just the tip of the iceberg.

Based on some discussions I've had lately, I suspect someone will resurrect it Sometime Soon.

> We might not need qemu-system support just yet.

Our focus on `qemu-system-hexagon` is one of my team's primary goals right now.  If we'd have gotten it landed sooner maybe we could have dedicated more time to this pull req.

Regardless of buildbots - linux userspace QEMU is ideal for running tests like the ones defined in llvm-test-suite.  We could also use it to run the libc++/libc++abi test suites.  Or maybe we already are testing some of those w/QEMU user?

But as far as libc tests - I'm not quite sure whether QEMU's linux-userspace is the right thing to use.  Making each architecture port of the libc robust enough to be able to interact with any host architecture's kernel and still pass comprehensive libc tests - this might be a higher bar than desired.  But there's bound to be a subset of tests that still make sense.

**Pull request closed without merging**